File: QuickSort.cpp
#include<iostream>
using namespace std;
void QuickSort(int *a,int l,int r){
    if(l>=r)
        return ;
    int i=l,j=r;
    int pivot=a[l];
    while(i<j){
        while(i<j && a[j]>=pivot) j--;
        while(i<j && a[i]<=pivot) i++;
        if(i<j) swap(a[i],a[j]);
    }
    a[l]=a[i];
    a[i]=pivot;
    QuickSort(a,l,i-1);
    QuickSort(a,i+1,r);
}
int main(){
    int n;
    cin>>n;
    int a[n+1];
    for(int i=1;i<=n;i++)
        cin>>a[i];
    QuickSort(a,1,n);
    for(int i=1;i<=n;i++)
        cout<<a[i]<<' ';
    return 0;
}

File: QuickSort.go
package main
import (
	"fmt"
)
func QuickSort(a []int,l int,r int){
	if l>=r{
		return
	}
	i,j,pivot := l,r,a[l]
	for ;i<j;{
		for ;i<j&&a[j]>=pivot;{
			j--
		}
		for ;i<j&&a[i]<=pivot;{
			i++
		}
		if i<j {
			a[i],a[j] = a[j],a[i]
		}
	}
	a[l]=a[i]
	a[i]=pivot
	QuickSort(a,l,i-1)
	QuickSort(a,i+1,r)
}
func main(){
	var a []int
	var n,x int
	fmt.Scanln(&n)
	for i:=0;i<n;i++{
		fmt.Scan(&x)
		a=append(a,x)
	}
	QuickSort(a,0,n-1)
	fmt.Println(a)
}